packages/yew-router/src/service.rs

```diff
@@ -1,182 +1,39 @@
-        let (history, location) = {
-            let window = web_sys::window().unwrap();
-            (
-                window
-                    .history()
-                    .expect("browser does not support history API"),
-                window.location(),
-            )
-        };
-
-        RouteService {
-            history,
-            location,
-            event_listener: None,
-            phantom_data: PhantomData,
-        }
-    }
-
-    #[inline]
-    fn get_route_from_location(location: &Location) -> String {
-        let path = location.pathname().unwrap();
-        let query = location.search().unwrap();
-        let fragment = location.hash().unwrap();
-        format_route_string(&path, &query, &fragment)
-    }
-
-    /// Gets the path name of the current url.
-    pub fn get_path(&self) -> String {
-        self.location.pathname().unwrap()
-    }
-
-    /// Gets the query string of the current url.
-    pub fn get_query(&self) -> String {
-        self.location.search().unwrap()
-    }
-
-    /// Gets the fragment of the current url.
-    pub fn get_fragment(&self) -> String {
-        self.location.hash().unwrap()
-    }
-}
-
-impl<STATE> RouteService<STATE>
-where
-    STATE: RouteState,
-{
-    /// Registers a callback to the route service.
-    /// Callbacks will be called when the History API experiences a change such as
-    /// popping a state off of its stack when the forward or back buttons are pressed.
-    pub fn register_callback(&mut self, callback: Callback<Route<STATE>>) {
-        let cb = move |event: PopStateEvent| {
-            let state_value: Value = event.state();
-            let state_string: String = state_value.as_string().unwrap_or_default();
-            let state: STATE = serde_json::from_str(&state_string).unwrap_or_else(|_| {
-                log::error!("Could not deserialize state string");
-                STATE::default()
-            });
-
-            // Can't use the existing location, because this is a callback, and can't move it in
-            // here.
-            let location: Location = web_sys::window().unwrap().location();
-            let route: String = Self::get_route_from_location(&location);
-
-            callback.emit(Route { route, state })
-        };
-
-        self.event_listener = Some(EventListener::new(
-            web_sys::window().unwrap().as_ref(),
-            "popstate",
-            move |event| {
-                let event: PopStateEvent = event.clone().dyn_into().unwrap();
-                cb(event)
-            },
-        ));
-    }
-
-    /// Sets the browser's url bar to contain the provided route,
-    /// and creates a history entry that can be navigated via the forward and back buttons.
+use crate::utils::build_path_with_base;
+use crate::Routable;
+use std::collections::HashMap;
+use wasm_bindgen::JsValue;
+use web_sys::Event;
+
+/// Service to interface with the router.
+#[derive(Debug, Copy, Clone)]
+pub struct RouterService;
```

Comment:
I think in that case, we should just have `yew_router::push()`. Having a module for service doesn't feel right.

Reply:
Totally agree with that. Not sure if `push` is the most obvious name though.

Reply:
What should it be called then?

Reply:
Actually, let's leave it in the `service` module for now. Because there's also `query` function in the service. We can move it out once there's a better way of handling query parameters